src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Sql/Sprocs/AddInstanceV6.sql

```diff
@@ -80,118 +80,147 @@ AS
-
-        INSERT INTO dbo.Study
-            (PartitionKey, StudyKey, StudyInstanceUid, PatientId, PatientName, PatientBirthDate, ReferringPhysicianName, StudyDate, StudyDescription, AccessionNumber)
-        VALUES
-            (@partitionKey, @studyKey, @studyInstanceUid, @patientId, @patientName, @patientBirthDate, @referringPhysicianName, @studyDate, @studyDescription, @accessionNumber)
-    END TRY
-    BEGIN CATCH
-        --Dupe on StudyInstanceUid, handle it in the code and retry once, instead of holding Serialization/HOLDLOCK on the range of study
-        IF ERROR_NUMBER() = 2601
-            -- Latest wins
-            UPDATE dbo.Study
-            SET PatientId = @patientId, PatientName = @patientName, PatientBirthDate = @patientBirthDate, ReferringPhysicianName = @referringPhysicianName, StudyDate = @studyDate, StudyDescription = @studyDescription, AccessionNumber = @accessionNumber
-            WHERE StudyKey = @studyKey
-        ELSE
-            THROW
-    END CATCH
-    ELSE
-    BEGIN
-        -- Latest wins
-        UPDATE dbo.Study
-        SET PatientId = @patientId, PatientName = @patientName, PatientBirthDate = @patientBirthDate, ReferringPhysicianName = @referringPhysicianName, StudyDate = @studyDate, StudyDescription = @studyDescription, AccessionNumber = @accessionNumber
-        WHERE StudyKey = @studyKey
-    END
-
-    -- Insert Series
-    SELECT @seriesKey = SeriesKey
-    FROM dbo.Series WITH(UPDLOCK)
-    WHERE StudyKey = @studyKey
-    AND SeriesInstanceUid = @seriesInstanceUid
-    AND PartitionKey = @partitionKey
-
-    IF @@ROWCOUNT = 0
-    BEGIN
-        SET @seriesKey = NEXT VALUE FOR dbo.SeriesKeySequence
-
-        INSERT INTO dbo.Series
-            (PartitionKey, StudyKey, SeriesKey, SeriesInstanceUid, Modality, PerformedProcedureStepStartDate, ManufacturerModelName)
-        VALUES
-            (@partitionKey, @studyKey, @seriesKey, @seriesInstanceUid, @modality, @performedProcedureStepStartDate, @manufacturerModelName)
-    END
-    ELSE
-    BEGIN
-        -- Latest wins
-        UPDATE dbo.Series
-        SET Modality = @modality, PerformedProcedureStepStartDate = @performedProcedureStepStartDate, ManufacturerModelName = @manufacturerModelName
-        WHERE SeriesKey = @seriesKey
-        AND StudyKey = @studyKey
-        AND PartitionKey = @partitionKey
-    END
-
-    -- Insert Instance
-    INSERT INTO dbo.Instance
-        (PartitionKey, StudyKey, SeriesKey, InstanceKey, StudyInstanceUid, SeriesInstanceUid, SopInstanceUid, Watermark, Status, LastStatusUpdatedDate, CreatedDate, TransferSyntaxUid)
-    VALUES
-        (@partitionKey, @studyKey, @seriesKey, @instanceKey, @studyInstanceUid, @seriesInstanceUid, @sopInstanceUid, @newWatermark, @initialStatus, @currentDate, @currentDate, @transferSyntaxUid)
+    -- We turn off XACT_ABORT so that we can rollback and retry the INSERT/UPDATE into the study table on failure
+    SET XACT_ABORT OFF
```

Comment:
This diff looks bizarre in GitHub, but the crux of the change is that the entire `TRANSACTION` is wrapped in a `TRY...CATCH` block because we have `SET XACT_ABORT OFF`. And the setting is `OFF` so that we can implicitly rollback the possibly bad `INSERT` into the `dbo.Study` table and instead use a `MERGE`